Remove Structures definitions in favor of binary-parser module

kylestev opened the pull request:
> This yields a sizable performance increase (10x) for a jar that is 969 KB in size and had ~180-190 classes parsing it in 0.55s on average.

@TSedlar what do you think of this? Thoughts?
- [`src/bytecode/parsers/BinaryParser.js`](https://github.com/kylestev/js-bytecode-core/blob/ca31f438003004536dbc93ff1adb71ee133b8e64/src/bytecode/parsers/BinaryParser.js)
- [`src/bytecode/Loader.js`](https://github.com/kylestev/js-bytecode-core/blob/ca31f438003004536dbc93ff1adb71ee133b8e64/src/bytecode/Loader.js)

kylestev commented:
I should also mention this has been tested on Java 6 and Java 8.

TSedlar commented:
@kylestev Looks incredibly sweet. I just suggest making both [=== 5 and === 6](https://github.com/kylestev/js-bytecode-core/blob/ca31f438003004536dbc93ff1adb71ee133b8e64/src/bytecode/parsers/BinaryParser.js#L35) constants.

kylestev committed:
refactor JVMAttribute classes

Since most of these attributes will use the byte array
in attrInfo.info, it makes sense to just throw this in the
constructor and let them access it in the decode method
in whatever manner they see fit for their decoding process.

kylestev committed:
add binary-parser package and jvm parsers

This yields a pretty fantastic performance boost and
removes the old Structures json-refs format that was
previously used for parsing elements in the JVM class
file format.
